src/live.rs

```diff
@@ -695,6 +696,37 @@ fn collect_iso_records(
     Ok(())
 }
 
+pub fn iso_extract_pxe(config: &IsoExtractPxeConfig) -> Result<()> {
+    let mut iso = IsoFs::new(&config.input)?;
+    let pxeboot = match iso.get_path(COREOS_ISO_PXEBOOT_DIR)? {
+        iso9660::DirectoryRecord::Directory(d) => d,
+        iso9660::DirectoryRecord::File(_) => {
+            bail!("expected dir {}, but found file", COREOS_ISO_PXEBOOT_DIR)
+        }
+    };
+
+    let outdir = config.output_dir.as_deref().unwrap_or(".");
+    std::fs::create_dir_all(outdir)?;
+
+    for record in iso.list_dir(&pxeboot)? {
+        match record? {
+            iso9660::DirectoryRecord::Directory(_) => continue,
+            iso9660::DirectoryRecord::File(file) => {
+                let basename = file.name.trim_end_matches(";1").trim_end_matches('.');
+                let path = Path::new(outdir).join(basename);
```

Comment:
A `file.name` of `../INUZ` will get through validation and perform a directory traversal.  The consequences seem limited because of the `create_new()`, but it'd still be good to fix.

Reply:
https://github.com/coreos/coreos-installer/pull/620/files#r708383059